### DISCUSSION TITLE
Repositionable UI elements

### DISCUSSION THREAD

**Discussion by PokkeTigrex** [FakeFishGames/Barotrauma#9026] [Ideas and suggestions]
**The issue**
Baro's UI has been rather finicky when it comes to multiple UI elements being on the screen, such as linked devices and inventory, or sonars and other devices with many pieces of UI. Issues like overlapping and odd positioning is frequent when interacting with structures linked to two or more other devices, which can cause the UI to freak out and get stuck in bad positions.

![image](https://user-images.githubusercontent.com/85914375/166122892-4e48ef00-bb7b-4052-b778-5da08df87d9d.png)
_Large Storage Locker linked with two Fabricators causes overlapping, often making certain parts nonfunctional due to being behind._

![image](https://user-images.githubusercontent.com/85914375/166122943-994ff3e9-fc1a-4fd8-b3ef-b3782a5179c3.png)
_Interacting with the fabricator linked to the Large Storage Locker after interacting with the Locker will also retain the position of UI for the fabricator which again, is almost impossible to use because of it overlapping._

This isn't the only issue with the UI, whereas having certain resolutions can cause the UI to overlap with other parts, an example being a standalone Sonar in a sub.

![image](https://user-images.githubusercontent.com/85914375/166123077-67e7b40c-f993-42d4-85ff-e5bb23ab4a58.png)
_The sonar in the middle of the Berilla being used, but cannot ping or use any other functions because of the sonar UI overlapping above the options and settings for it while using 1280x960 in fullscreen._

**A possible solution for ingame**
Make basically all (or at least most) interactable UI elements draggable. This basically allows the player to fix both issue themselves by moving it into a more favorable position of their liking. A pin feature can make it so the element is locked and cannot be moved without unpinning it, or by whatever tries to reset or reposition the UI automatically, and possible retain the position it between sessions. This would basically a fix (if implementation doesn't break stuff) and a QoL change to the UI. This obviously won't fix resolution menu UI overlapping that I've seen reported, so that would be a separate thing all together, this is just good enough for in-game play.

**Comment by PokkeTigrex** [comment]
Posted an issue up as well as it is still a bug/design issue https://github.com/Regalis11/Barotrauma/issues/9032.

**Comment by Ek-Videogames** [comment]
This could even benefit in single-GUI situations such as just being able to move around the handheld-sonar window while equipped so it doesn't have to be blocking the lower left quadrant of the screen from mouse interaction.

Perhaps a "grip" similar to how it is implemented on movable container GUIs.
